Replaced boost::asio::io_service with boost::asio::io_context

Bekreth opened the pull request:
MSPDS was unable to build due to deprecated boost functionality.  This PR adds a patch file for a 1-to-1 replacement of `io_service` to `io_context` using non-deprecated functions.

## Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- Built on platform:
  - [X] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- Tested, as applicable:
  - [ ] [NixOS tests] in [nixos/tests].
  - [ ] [Package tests] at `passthru.tests`.
  - [ ] Tests in [lib/tests] or [pkgs/test] for functions and "core" functionality.
- [ ] Ran `nixpkgs-review` on this PR. See [nixpkgs-review usage].
- [X] Tested basic functionality of all binary files, usually in `./result/bin/`.
  - Ran `mspdebug tilib` locally to verify that `mspdebug` was able to locate the built `libmsp430.so` file and connect via USB to a TI MSP-EXP430G2ET.  Verified as well that the `libmsp430.so` file was able to successfully update the firmware of the development board
- Nixpkgs Release Notes
  - [ ] Package update: when the change is major or breaking.
- NixOS Release Notes
  - [ ] Module addition: when adding a new NixOS module.
  - [ ] Module update: when the change is significant.
- [ ] Fits [CONTRIBUTING.md], [pkgs/README.md], [maintainers/README.md] and other READMEs.

[NixOS tests]: https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests
[Package tests]: https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests
[nixpkgs-review usage]: https://github.com/Mic92/nixpkgs-review#usage

[CONTRIBUTING.md]: https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md
[lib/tests]: https://github.com/NixOS/nixpkgs/blob/master/lib/tests
[maintainers/README.md]: https://github.com/NixOS/nixpkgs/blob/master/maintainers/README.md
[nixos/tests]: https://github.com/NixOS/nixpkgs/blob/master/nixos/tests
[pkgs/README.md]: https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md
[pkgs/test]: https://github.com/NixOS/nixpkgs/blob/master/pkgs/test

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

Bekreth commented on pkgs/development/misc/msp430/mspds/boost.patch:
```diff
@@ -0,0 +1,131 @@
+ 	boost::system::error_code ec;
+ 
+-	while (ioService->run_one(ec))
++	while (ioContext->run_one())
```
Per boost documentation (https://www.boost.org/doc/libs/1_70_0/doc/html/boost_asio/reference/io_context/run.html) the non-error_code overload should be preferred

acid-bong commented:
OP, note that we have local title conventions (apply for both commits and PRs). The correct format should be `PROGRAM: CHANGES`, if you're editing a package, or `nixos/PROGRAM: CHANGES`, if it's a NixOS module.

See more in the contributing docs.

May `git commit --amend`, `git rebase -i` and `git push --force-with-lease` be with you 🖖

Bekreth committed:
mspds: Replaced boost::asio::io_service with boost::asio::io_context

Updates to Boost have rendered this package unbuildable.  This update patches the existing source to get the module functional again
